From: Daniel Baker <dbaker@mozilla.com>
Date: Fri, 15 Mar 2024 15:30:10 -0600
Subject: (tmp-cherry-pick) Revert "Delete pc/peerconnection build target"
 (771b524606)

This reverts commit 18a42e3272a6a25a23042fd39e67de02def8cafb.

Reason for revert: Breaks downstream project.

Original change's description:
> Delete pc/peerconnection build target
>
> It is not useful any more.
>
> Bug: webrtc:13634, b/238176207
> Change-Id: I3dd4ebca355bb828c6c3c30392333d9fe03a478c
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/267821
> Commit-Queue: Harald Alvestrand <hta@webrtc.org>
> Reviewed-by: Mirko Bonadei <mbonadei@webrtc.org>
> Cr-Commit-Position: refs/heads/main@{#41427}

Bug: webrtc:13634, b/238176207
Change-Id: Ib53e0b0cc81ac218e3c19e4c652ffe0b19155c22
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/332220
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Owners-Override: Christoffer Dewerin <jansson@google.com>
Commit-Queue: Christoffer Dewerin <jansson@google.com>
Cr-Commit-Position: refs/heads/main@{#41430}
---
 pc/BUILD.gn          | 142 +++++++++++++++++++++++++++++++++++++++++++
 sdk/android/BUILD.gn |   1 +
 2 files changed, 143 insertions(+)

diff --git a/pc/BUILD.gn b/pc/BUILD.gn
index e351748485..baab6f715e 100644
--- a/pc/BUILD.gn
+++ b/pc/BUILD.gn
@@ -16,6 +16,7 @@
 # - rtc_pc
 # - session_description
 # - simulcast_description
+# - peerconnection
 # - sdp_utils
 # - media_stream_observer
 # - video_track_source
@@ -735,6 +736,142 @@ rtc_library("media_protocol_names") {
   absl_deps = [ "//third_party/abseil-cpp/absl/strings" ]
 }
 
+rtc_source_set("peerconnection") {
+  # TODO(bugs.webrtc.org/13661): Reduce visibility if possible
+  visibility = [ "*" ]  # Used by Chromium and others
+  allow_poison = [ "environment_construction" ]
+  cflags = []
+  sources = []
+
+  deps = [
+    ":audio_rtp_receiver",
+    ":audio_track",
+    ":connection_context",
+    ":data_channel_controller",
+    ":data_channel_utils",
+    ":dtmf_sender",
+    ":ice_server_parsing",
+    ":jitter_buffer_delay",
+    ":jsep_ice_candidate",
+    ":jsep_session_description",
+    ":legacy_stats_collector",
+    ":legacy_stats_collector_interface",
+    ":local_audio_source",
+    ":media_protocol_names",
+    ":media_stream",
+    ":media_stream_observer",
+    ":peer_connection",
+    ":peer_connection_factory",
+    ":peer_connection_internal",
+    ":peer_connection_message_handler",
+    ":proxy",
+    ":remote_audio_source",
+    ":rtc_stats_collector",
+    ":rtc_stats_traversal",
+    ":rtp_parameters_conversion",
+    ":rtp_receiver",
+    ":rtp_sender",
+    ":rtp_transceiver",
+    ":rtp_transmission_manager",
+    ":sctp_data_channel",
+    ":sdp_offer_answer",
+    ":sdp_state_provider",
+    ":sdp_utils",
+    ":session_description",
+    ":simulcast_description",
+    ":simulcast_sdp_serializer",
+    ":stream_collection",
+    ":track_media_info_map",
+    ":transceiver_list",
+    ":usage_pattern",
+    ":video_rtp_receiver",
+    ":video_track",
+    ":video_track_source",
+    ":webrtc_sdp",
+    ":webrtc_session_description_factory",
+    "../api:array_view",
+    "../api:async_dns_resolver",
+    "../api:audio_options_api",
+    "../api:call_api",
+    "../api:fec_controller_api",
+    "../api:field_trials_view",
+    "../api:frame_transformer_interface",
+    "../api:ice_transport_factory",
+    "../api:libjingle_logging_api",
+    "../api:libjingle_peerconnection_api",
+    "../api:media_stream_interface",
+    "../api:network_state_predictor_api",
+    "../api:packet_socket_factory",
+    "../api:priority",
+    "../api:rtc_error",
+    "../api:rtc_event_log_output_file",
+    "../api:rtc_stats_api",
+    "../api:rtp_parameters",
+    "../api:rtp_transceiver_direction",
+    "../api:scoped_refptr",
+    "../api:sequence_checker",
+    "../api/adaptation:resource_adaptation_api",
+    "../api/audio_codecs:audio_codecs_api",
+    "../api/crypto:frame_decryptor_interface",
+    "../api/crypto:options",
+    "../api/neteq:neteq_api",
+    "../api/rtc_event_log",
+    "../api/task_queue",
+    "../api/task_queue:pending_task_safety_flag",
+    "../api/transport:bitrate_settings",
+    "../api/transport:datagram_transport_interface",
+    "../api/transport:enums",
+    "../api/transport:field_trial_based_config",
+    "../api/transport:network_control",
+    "../api/transport:sctp_transport_factory_interface",
+    "../api/units:data_rate",
+    "../api/video:builtin_video_bitrate_allocator_factory",
+    "../api/video:video_bitrate_allocator_factory",
+    "../api/video:video_codec_constants",
+    "../api/video:video_frame",
+    "../api/video:video_rtp_headers",
+    "../api/video_codecs:video_codecs_api",
+    "../call:call_interfaces",
+    "../call:rtp_interfaces",
+    "../call:rtp_sender",
+    "../common_video",
+    "../logging:ice_log",
+    "../media:rtc_data_sctp_transport_internal",
+    "../media:rtc_media_base",
+    "../media:rtc_media_config",
+    "../modules/audio_processing:audio_processing_statistics",
+    "../modules/rtp_rtcp:rtp_rtcp_format",
+    "../p2p:rtc_p2p",
+    "../rtc_base:callback_list",
+    "../rtc_base:checks",
+    "../rtc_base:ip_address",
+    "../rtc_base:network_constants",
+    "../rtc_base:rtc_operations_chain",
+    "../rtc_base:safe_minmax",
+    "../rtc_base:socket_address",
+    "../rtc_base:threading",
+    "../rtc_base:weak_ptr",
+    "../rtc_base/experiments:field_trial_parser",
+    "../rtc_base/network:sent_packet",
+    "../rtc_base/synchronization:mutex",
+    "../rtc_base/system:file_wrapper",
+    "../rtc_base/system:no_unique_address",
+    "../rtc_base/system:rtc_export",
+    "../rtc_base/system:unused",
+    "../rtc_base/third_party/base64",
+    "../rtc_base/third_party/sigslot",
+    "../stats",
+    "../system_wrappers",
+    "../system_wrappers:field_trial",
+    "../system_wrappers:metrics",
+  ]
+  absl_deps = [
+    "//third_party/abseil-cpp/absl/algorithm:container",
+    "//third_party/abseil-cpp/absl/strings",
+    "//third_party/abseil-cpp/absl/types:optional",
+  ]
+}
+
 rtc_library("sctp_data_channel") {
   visibility = [ ":*" ]
   sources = [
@@ -1982,6 +2119,7 @@ if (rtc_include_tests && !build_with_chromium) {
       ":media_protocol_names",
       ":media_session",
       ":pc_test_utils",
+      ":peerconnection",
       ":rtc_pc",
       ":rtcp_mux_filter",
       ":rtp_media_utils",
@@ -2082,6 +2220,7 @@ if (rtc_include_tests && !build_with_chromium) {
     deps = [
       ":pc_test_utils",
       ":peer_connection",
+      ":peerconnection",
       ":peerconnection_wrapper",
       "../api:audio_options_api",
       "../api:create_peerconnection_factory",
@@ -2134,6 +2273,7 @@ if (rtc_include_tests && !build_with_chromium) {
     ]
     deps = [
       ":pc_test_utils",
+      ":peerconnection",
       ":sdp_utils",
       "../api:function_view",
       "../api:libjingle_peerconnection_api",
@@ -2491,6 +2631,7 @@ if (rtc_include_tests && !build_with_chromium) {
       ":peer_connection",
       ":peer_connection_factory",
       ":peer_connection_proxy",
+      ":peerconnection",
       ":remote_audio_source",
       ":rtp_media_utils",
       ":rtp_parameters_conversion",
@@ -2649,6 +2790,7 @@ if (rtc_include_tests && !build_with_chromium) {
       ":jitter_buffer_delay",
       ":libjingle_peerconnection",
       ":peer_connection_internal",
+      ":peerconnection",
       ":rtp_receiver",
       ":rtp_sender",
       ":sctp_data_channel",
diff --git a/sdk/android/BUILD.gn b/sdk/android/BUILD.gn
index 1f672662b7..88fd1e712c 100644
--- a/sdk/android/BUILD.gn
+++ b/sdk/android/BUILD.gn
@@ -828,6 +828,7 @@ if (current_os == "linux" || is_android) {
       ":generated_metrics_jni",
       ":native_api_jni",
       ":peerconnection_jni",
+      "../../pc:peerconnection",
       "../../rtc_base:stringutils",
       "../../system_wrappers:metrics",
     ]
